internal/factsengine/gatherers/osrelease.go

```diff
@@ -0,0 +1,83 @@
+package gatherers
+
+import (
+	"os"
+
+	"github.com/hashicorp/go-envparse"
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+)
+
+const (
+	OSReleaseGathererName = "os-release"
+	OSReleaseFilePath     = "/etc/os-release"
+)
+
+// nolint:gochecknoglobals
+var (
+	OSReleaseFileError = entities.FactGatheringError{
+		Type:    "os-release-file-error",
+		Message: "error reading /etc/os-release file",
+	}
+
+	OSReleaseDecodingError = entities.FactGatheringError{
+		Type:    "os-release-decoding-error",
+		Message: "error decoding file content",
+	}
+)
+
+type OSReleaseGatherer struct {
+	osReleaseFilePath string
+}
+
+func NewDefaultOSReleaseGatherer() *OSReleaseGatherer {
+	return NewOSReleaseGatherer(OSReleaseFilePath)
+}
+
+func NewOSReleaseGatherer(path string) *OSReleaseGatherer {
+	return &OSReleaseGatherer{
+		osReleaseFilePath: path,
+	}
+}
+
+func (g *OSReleaseGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting %s facts gathering process", OSReleaseGathererName)
+
+	file, err := os.Open(g.osReleaseFilePath)
+	if err != nil {
+		log.Error(err)
+		return facts, OSReleaseFileError.Wrap(err.Error())
+	}
+	defer file.Close()
```

Comment:
Let's puts some logs in this `defer`. Just in case. 
As we do in: https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/groups.go#L61